web/src/stream_events.ts

```diff
@@ -104,12 +104,10 @@ export function update_property<P extends keyof UpdatableStreamProperties>(
             sub,
             group_setting_value_schema.parse(value),
         );
-        const $elem = $(
-            `#id_group_permission_${CSS.escape(sub.stream_id.toString())}_${CSS.escape(property)}`,
-        );
-        user_group_edit.update_setting_in_group_permissions_panel(
-            $elem,
+        user_group_edit.update_stream_setting_in_permissions_panel(
```

Comment:
Using different functions for realm, group and stream settings as otherwise would need a long if-else blocks to set fields as per the type of setting and that made code somewhat difficult to read.